presentation-compiler/src/main/dotty/tools/pc/DiagnosticProvider.scala

```diff
@@ -0,0 +1,70 @@
+package dotty.tools.pc
+
+import org.eclipse.lsp4j
+import org.eclipse.lsp4j.DiagnosticSeverity
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.interfaces.Diagnostic as DiagnosticInterfaces
+import dotty.tools.dotc.reporting.Diagnostic
+import dotty.tools.pc.utils.InteractiveEnrichments.toLsp
+
+import scala.meta.pc.VirtualFileParams
+import dotty.tools.dotc.reporting.CodeAction
+import dotty.tools.dotc.rewrites.Rewrites.ActionPatch
+import scala.jdk.CollectionConverters.*
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.reporting.ErrorMessageID
+import org.eclipse.lsp4j.DiagnosticTag
+import org.eclipse.lsp4j.CodeActionKind
```

Comment:
Maybe sort the imports if possible? Metals should allow it

Reply:
https://github.com/scalameta/metals/pull/7915 <- that might help, we can try later with a snapshot